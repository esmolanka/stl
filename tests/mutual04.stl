
type Either a b = < Left : a, Right : b >

mutual (a : Type)
  | Even = < EZero : a, ESucc : Odd >
  | Odd  = < OZero : a, OSucc : Even >
  | Foo  = forall (a : Nat). Nat a -> Either Even Odd

#eval Foo
