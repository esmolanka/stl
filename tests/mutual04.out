tests/mutual04.stl:9:1-9:10:
  Normalised:
    λ a. ∀ (a : Nat).
      (Nat a)
      -> ∀ (ρ : Row). ∃ (ω : #). ∃ (ω : #).
        < Left^ω :
            μ Even. ∀ (ρ : Row). ∃ (ω : #). ∃ (ω : #).
              < EZero^ω : a/1
              | ESucc^ω/1 :
                  ∀ (ρ : Row). ∃ (ω : #). ∃ (ω : #).
                    <OZero^ω : a/1| OSucc^ω/1 : Even | ... ρ>
              | ... ρ
              >
        | Right^ω/1 :
            μ Odd. ∀ (ρ : Row). ∃ (ω : #). ∃ (ω : #).
              < OZero^ω : a/1
              | OSucc^ω/1 :
                  ∀ (ρ : Row). ∃ (ω : #). ∃ (ω : #).
                    <EZero^ω : a/1| ESucc^ω/1 : Odd | ... ρ>
              | ... ρ
              >
        | ... ρ
        >
  Kind:
    Type -> Type